onebusaway-android/src/google/java/org/onebusaway/android/map/googlemapsv2/bike/BikeStationOverlay.java

```diff
@@ -0,0 +1,123 @@
+package org.onebusaway.android.map.googlemapsv2.bike;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.drawable.Drawable;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import org.onebusaway.android.R;
+import org.onebusaway.android.app.Application;
+import org.onebusaway.android.map.googlemapsv2.BaseMapFragment;
+import org.onebusaway.android.map.googlemapsv2.MapHelpV2;
+import org.onebusaway.android.map.googlemapsv2.MarkerListeners;
+import org.opentripplanner.routing.bike_rental.BikeRentalStation;
+
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Class to hold bike stations and control their display on the map.
+ *
+ * @author carvalhorr
+ */
+public class BikeStationOverlay implements MarkerListeners {
+
+    private GoogleMap mMap;
+
+    private static final int FUZZY_MAX_MARKER_COUNT = 200;
+
+    private HashMap<Marker, BikeRentalStation> mStations;
+
+    private BaseMapFragment.OnFocusChangedListener mOnFocusChangedListener;
+
+    private BitmapDescriptor mSmallBikeStationIcon;
+
+    private BitmapDescriptor mBigBikeStationIcon;
+
+    public BikeStationOverlay(Activity a, GoogleMap map) {
+        mMap = map;
+        mStations = new HashMap<>();
+        mMap.setInfoWindowAdapter(new BikeInfoWindow(a));
+
+        mSmallBikeStationIcon = BitmapDescriptorFactory.fromBitmap(createBitmapFromShape());
+        mBigBikeStationIcon = BitmapDescriptorFactory.fromResource(R.drawable.bike_marker_big);
+    }
+
+    private synchronized void addMarker(BikeRentalStation station) {
+        MarkerOptions options = new MarkerOptions().position(MapHelpV2.makeLatLng(station.y, station.x));
+        if (mMap.getCameraPosition().zoom > 13) {
+            options.icon(mBigBikeStationIcon);
+        } else {
+            options.icon(mSmallBikeStationIcon);
+        }
+        Marker m = mMap.addMarker(options);
+
+        m.setTag(station);
```

Comment:
So unfortunately  `Marker.setTag()` isn't available in the Amazon Maps API v2 library:
https://travis-ci.org/OneBusAway/onebusaway-android#L2089

It's a newer feature of Google Maps API v2.

To keep compatibility with Amazon Maps API v2, you can just pull the bikes out of `mStations` instead, I believe.  This is how `StopOverlay` is implemented, as `get/setTag()` wasn't implemented in Google Maps API v2 when I implemented `StopOverlay`.